lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1360,36 +1362,85 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
-                  "expected weight and input tensor to have static shape");
-            const int64_t dilatedKernelSize =
-                dilations[dimIdx] * (weightShape[dimIdx + 2] - 1) + 1;
-            int64_t totalPad = ((inputShape[dimIdx + 2] + strides[dimIdx] - 1) /
-                                    strides[dimIdx] -
-                                1) *
-                                   strides[dimIdx] +
-                               dilatedKernelSize - inputShape[dimIdx + 2];
-            totalPad = totalPad >= 0 ? totalPad : 0;
-            padding[dimIdx] =
-                isSameLower ? ((totalPad + 1) / 2) : (totalPad / 2);
-            padding[spatialRank + dimIdx] = totalPad - padding[dimIdx];
+            // dilatedSize = dilations[dimIdx]*(weightShape[dimIdx + 2] - 1) + 1
+            Value cst1 = rewriter.create<Torch::ConstantIntOp>(loc, 1);
+            Value dilationValue =
+                rewriter.create<Torch::ConstantIntOp>(loc, dilations[dimIdx]);
+            Value weightDimSize =
+                Torch::getTensorDimSize(rewriter, weight, dimIdx + 2);
+            Value weightMinus =
+                rewriter.create<Torch::AtenSubIntOp>(loc, weightDimSize, cst1);
+            Value dilationWeight = rewriter.create<Torch::AtenMulIntOp>(
+                loc, dilationValue, weightMinus);
+            Value dilatedKernelSize =
+                rewriter.create<Torch::AtenAddIntOp>(loc, dilationWeight, cst1);
+
+            // totalPad = (((inputShape[dimIdx + 2] + strides[dimIdx] -1) /
+            //              strides[dimIdx]) - 1) * strides[dimIdx] +
+            //              dilatedKernelSize - inputShape[dimIdx + 2];
+
+            Value stridesValue =
+                rewriter.create<Torch::ConstantIntOp>(loc, strides[dimIdx]);
+            Value inputDimSize =
+                Torch::getTensorDimSize(rewriter, input, dimIdx + 2);
+            Value stridesMinus =
+                rewriter.create<Torch::AtenSubIntOp>(loc, stridesValue, cst1);
+            Value inputStrides = rewriter.create<Torch::AtenAddIntOp>(
+                loc, inputDimSize, stridesMinus);
+            inputStrides = rewriter.create<Torch::AtenFloordivIntOp>(
+                loc, inputStrides, stridesValue);
+            inputStrides =
+                rewriter.create<Torch::AtenSubIntOp>(loc, inputStrides, cst1);
+            inputStrides = rewriter.create<Torch::AtenMulIntOp>(
+                loc, inputStrides, stridesValue);
+            Value strideDilatedSize = rewriter.create<Torch::AtenAddIntOp>(
+                loc, inputStrides, dilatedKernelSize);
+            Value totalPad = rewriter.create<Torch::AtenSubIntOp>(
+                loc, strideDilatedSize, inputDimSize);
+
+            // totalPad = totalPad > 0 ? totalPad : 0;
+            Value cst0 = rewriter.create<Torch::ConstantIntOp>(loc, 0);
+            totalPad =
+                rewriter.create<Torch::PrimMaxIntOp>(loc, totalPad, cst0);
```

Comment:
Although, you've not given random names to variable but I think you can improve them. For example:

`stridesMinus` -> `strideMinusOne`
`weightMinus` -> `weightMinusOne`
`dilationWeight` -> `dilationMulWeight`

Similarly, for all other places where it's needed.